utility_scripts/quota_audit.py

```diff
@@ -0,0 +1,171 @@
+from novaclient import client as novaclient
+from cinderclient.v2 import client as cinderclient
+
+
+def diff_moc_quotas(project_quotas):
+
+    different_quotas = {}
+    for resource in moc_standards:
+        if moc_standards[resource] == project_quotas[resource]:
+            continue
+        else:
+            different_quotas[resource] = project_quotas[resource]
+    return different_quotas
+
+
+def to_single_dict(nova_q, cinder_q, neutron_q):
+
+    combined = {}
+    combined.update(nova_q)
+    combined.update(cinder_q)
+    combined.update(neutron_q)
+    return combined
+
+
+def proj_to_request_dict(project_name):
+
+    request_dict = {}
+    for row in reader:
+        if row['OpenStack project name'] == project_name:
+            request_dict['instances'] = row['Instances']
+            request_dict['cores'] = row['VCPUs']
+            request_dict['ram'] = row['RAM']
+            request_dict['floatingip'] = row['Floating IPs']
+            request_dict['network'] = row['Networks']
+            request_dict['port'] = row['Ports']
+            request_dict['volumes'] = row['Volumes']
+            request_dict['snapshots'] = row['Snapshots']
+            request_dict['gigabytes'] = row['Volume & Snapshot Storage']
+    return request_dict
+
+
+def isolate_requests(dict_with_blanks):  # singles out the requests
+
+    configured_dict = {k: v for k, v in dict_with_blanks.items() if v}
+    if 'ram' in configured_dict.keys():
+        configured_dict['ram'] = str(int(configured_dict['ram']) * 1024)
+    return configured_dict
+
+
+def compare_request_with_real(requested_quotas, all_quotas):
+    
+    for key in requested_quotas:
+        if int(requested_quotas[key]) == all_quotas[key]:
+            continue
+        else:
+            return False
+    return True
+
+
+parser = argparse.ArgumentParser()
+parser.add_argument('filename')
+parser.add_argument('config', nargs='?')
+args = parser.parse_args()
+
+if args.config:
+    CONFIG_FILE = args.config
+    config = ConfigParser.ConfigParser()
+    config.read(CONFIG_FILE)
+    admin_user = config.get('auth', 'admin_user')
+    admin_pwd = config.get('auth', 'admin_pwd')
+    admin_project = config.get('auth', 'admin_project')
+    auth_url = config.get('auth', 'auth_url')
+else:
+    admin_user = os.environ.get('OS_USERNAME')
+    admin_pwd = os.environ.get('OS_PASSWORD')
+    admin_project = os.environ.get('OS_TENANT_NAME')
+    auth_url = os.environ.get('OS_AUTH_URL')
+
+
+auth = v2.Password(auth_url=auth_url,
+                   username=admin_user,
+                   password=admin_pwd,
+                   tenant_name=admin_project)
+
+sess = session.Session(auth=auth)
+keystone = client.Client(session=sess)
+ks_projects = keystone.projects.list()
+neutron = nclient.Client(session=sess)
+nova = novaclient.Client(2, session=sess)
+cinder = cinderclient.Client(session=sess)
+
+with open(args.filename, "rb") as source:
+    reader = list(csv.DictReader(source))
```

Comment:
Same deal here. I don't mind globals, but this ought to be `READER` so when we see it in `proj_to_request_dict` we have some idea of where it came from